routes/error-tracker.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2017 The AMP Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ **/
+
+/**
+ * objects used
+ * Objects are identical to the objects used in previous errortracker.go
+ * @typedef errorRequestMeta {
+	HTTPReferrer: string,
+	HTTPUserAgent: string
+    }
+ @typedef errorRequest {
+	URL:  string,
+	Method: string,
+	Meta: errorRequestMeta
+  }
+ @typedef errorEvent {
+	Application: string,
+	AppID; string,
+	Environment: string,
+	Version: string,
+	Message: string,
+	Exception: string,
+	Request: *errorRequest,
+	Filename: string,
+	Line: int32,
+	Classname: string,
+	Function: string,
+	Severity: string
+   }
+ **/
+
+/**
+ * Handle error requests from clients and log them.
+ **/
+
+const express = require('express');
+const logging = require('@google-cloud/logging');
+const winston = require('winston');
+const statusCodes = require('http-status-codes');
+const url = require('url');
+const router = express.Router();
+const appEngineProjectId = '';
+const logName = 'javascript.errors';
+const SERVER_START_TIME = Date.now();
+const filteredMessageOrException = ['stop_youtube',
+  'null%20is%20not%20an%20object%20(evaluating%20%27elt.parentNode%27)'];
+/**
+ * ERROR_LEVELS
+ * @enum {string}
+ */
+const ERROR_LEVELS = {
+  INFO: 'Info',
+  ERROR: 'Error',
+};
+const SEVERITY = {
+  INFO: 200,
+  ERROR: 500,
+};
+
+/**
+ * @param message
+ * @param exception
+ * @return {boolean}
+ */
+function isFilteredMessageOrException(message, exception) {
+  return filteredMessageOrException.some(function (msg) {
+    return message.includes(msg) ||
+      exception.includes(msg);
+  });
+}
+
+/**
+ * @desc handle errors that come from an attempt to write to the logs
+ * @param {!error} err
+ * @param res http response object
+ * @param req http request object
+ */
+function logWritingError(err, res, req) {
+  if (err) {
+    res.status(statusCodes.INTERNAL_SERVER_ERROR);
+    res.send({error: 'Cannot write to Google Cloud Logging'});
+    res.end();
+    winston.error(appEngineProjectId, 'Cannot write to Google Cloud Logging: '
+      + url.parse(req.url, true).query['v'], err);
+  }
+}
+
+/**
+ * @desc extract params in GET request from query and fill errorEvent object
+ * log level by default is INFO.
+ * @param req
+ * @param res
+ * @param next
+ */
+function getHandler(req, res, next) {
+  const params = req.query;
+  const referer = params.r;
+  let errorType = 'default';
+  let isUserError = false;
+  if (params.a === '1') {
+    errorType = 'assert';
+    isUserError = true;
+  }
+
+  let severity = SEVERITY.INFO;
+  /**
```

Comment:
This comment style is reserved for JSDoc comments only. Inline comments like this should use `//`.